dask_cuda/utils.py

```diff
@@ -32,14 +32,26 @@ def setup(self, worker=None):
 
 
 class RMMSetup:
-    def __init__(self, nbytes, managed_memory, log_directory):
+    def __init__(self, nbytes, managed_memory, async_alloc, log_directory):
         self.nbytes = nbytes
         self.managed_memory = managed_memory
+        self.async_alloc = async_alloc
         self.logging = log_directory is not None
         self.log_directory = log_directory
 
     def setup(self, worker=None):
-        if self.nbytes is not None or self.managed_memory is True:
+        if self.async_alloc is True:
```

Comment:
I added a separate conditional checking for the async allocator to make sure it is mutually exclusive with pools or managed memory - is there a cleaner way of doing this or should I not bother?

Reply:
I think we should either raise an error if the user specifies both (otherwise, given the assumption they're mutually exclusive, it's impossible for a user to know which one is actually being used), or at least emit a warning. Warnings can be annoying and easy to ignore, thus I'd prefer raising an error.

Reply:
Sounds good - I opted for a `ValueError` if both are enabled, so I ended up leaving the separate conditional since now it shouldn't be ambiguous which RMM setup is used.